### PR TITLE
Move TODO aliases to aliases.nix

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -803,11 +809,14 @@ mapAliases ({
...
   xpraGtk3 = xpra; # added 2018-09-13
   xv = xxv; # added 2020-02-22
+  yacc = bison; # moved from top-level 2021-03-14
   youtubeDL = youtube-dl;  # added 2014-10-26
   ytop = throw "ytop has been abandoned by upstream. Consider switching to bottom instead";
   yubikey-neo-manager = throw "yubikey-neo-manager has been removed because it was broken. Use yubikey-manager-qt instead."; # added 2021-03-08
   yuzu = yuzu-mainline; # added 2021-01-25
   zdfmediathk = mediathekview; # added 2019-01-19
+
+
```
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:819]:
```suggestion
```

File: pkgs/top-level/aliases.nix [hunk 2]
```diff
@@ -384,10 +388,12 @@ mapAliases ({
   lua5_expat = luaPackages.luaexpat; # added 2017-05-02
   lua5_sec = luaPackages.luasec; # added 2017-05-02
   lxappearance-gtk3 = throw "lxappearance-gtk3 has been removed. Use lxappearance instead, which now defaults to Gtk3";  # added 2020-06-03
+  xz = xz; # moved from top-level 2021-03-14
```
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:391]:
```suggestion
  lzma = xz; # moved from top-level 2021-03-14
```

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by grahamc** [comment]
In principle I +1 the removal of aliases in all-packages.nix, so looks good to me from that point. We should probably CC the maintainers of the ~9 packages which had aliases moved.

File: pkgs/applications/kde/akonadi/default.nix [hunk 3]
```diff
@@ -2,8 +2,8 @@
   mkDerivation, lib, kdepimTeam,
   extra-cmake-modules, shared-mime-info, qtbase, accounts-qt,
   boost, kaccounts-integration, kcompletion, kconfigwidgets, kcrash, kdbusaddons,
-  kdesignerplugin, ki18n, kiconthemes, kio, kitemmodels, kwindowsystem, mysql, qttools,
```
Comment (adisbladis) [inline pkgs/applications/kde/akonadi/default.nix:5]: You should cc the `mysql/mariadb/akonadi/kde` maintainers here to ask about this name change as this breaks `akonadio.override { mysql = foo; }`

File: pkgs/top-level/aliases.nix [hunk 4]
```diff
@@ -803,11 +809,13 @@ mapAliases ({
   xlibs = xorg; # added 2015-09
   xpraGtk3 = xpra; # added 2018-09-13
   xv = xxv; # added 2020-02-22
+  yacc = bison; # moved from top-level 2021-03-14
   youtubeDL = youtube-dl;  # added 2014-10-26
   ytop = throw "ytop has been abandoned by upstream. Consider switching to bottom instead";
   yubikey-neo-manager = throw "yubikey-neo-manager has been removed because it was broken. Use yubikey-manager-qt instead."; # added 2021-03-08
   yuzu = yuzu-mainline; # added 2021-01-25
   zdfmediathk = mediathekview; # added 2019-01-19
+
```
Comment (dotlambda) [inline pkgs/top-level/aliases.nix:818]: Why though?

File: pkgs/applications/misc/azuredatastudio/default.nix [hunk 5]
```diff
@@ -12,7 +12,7 @@
 , at-spi2-atk
 , gnutar
 , atomEnv
-, kerberos
```
Comment (adisbladis) [inline pkgs/applications/misc/azuredatastudio/default.nix:15]: Same here, cc the relevant maintainers..

File: pkgs/applications/misc/golden-cheetah/default.nix [hunk 6]
```diff
@@ -1,7 +1,7 @@
 { lib, fetchFromGitHub, fetchpatch, mkDerivation
 , qtbase, qtsvg, qtserialport, qtwebengine, qtmultimedia, qttools
 , qtconnectivity, qtcharts, libusb-compat-0_1
-, yacc, flex, zlib, qmake, makeDesktopItem, makeWrapper
```
Comment (adisbladis) [inline pkgs/applications/misc/golden-cheetah/default.nix:4]: And so on and so forth...

**Review by adisbladis — CHANGES_REQUESTED** [review]

File: pkgs/top-level/aliases.nix (hunk 4 above)
Comment (SuperSandro2000) [reply pkgs/top-level/aliases.nix:818]: The alphabetical sorting ends here. I can drop that if it is a real problem.

Comment (dotlambda) [reply pkgs/top-level/aliases.nix:818]: At least it doesn't seem to match any of the commit messages.

**Comment by SuperSandro2000** [comment]
pining the maintainers of the packages which aliases where moved.
gtest:
@Zoomulator @ivan-tkatchev (no github name)

iproute:
@primeos @eelco @fpletz @globin 

jack2:
@goibhniu 

libkrb5:
has no maintainers

xz:
@svanderburg

mariadb:
@thoughtpolice

bison:
has no maintainers

File: pkgs/top-level/aliases.nix (hunk 4 above)
Comment (dotlambda) [reply pkgs/top-level/aliases.nix:841]: Please add a separate commit for this change if you include it.

Comment (SuperSandro2000) [reply pkgs/top-level/aliases.nix:841]: This is really bikeshedding.

Comment (dotlambda) [reply pkgs/top-level/aliases.nix:841]: Alright, a better suggestion: Make one PR per alias. 237 changed files is too much to review.

Comment (SuperSandro2000) [reply pkgs/top-level/aliases.nix:841]: I did one alias per commit which should be easily reviewable.

**Comment by ajs124** [comment]
Since `iproute` wasn't replaced in all modules, nixpkgs cannot be evaluated without aliases right now.

**Comment by ajs124** [comment]
#118805

**Pull request merged**